sips/sip-022/sip-022-emergency-pox-fix.md

```diff
@@ -0,0 +1,235 @@
+        .iter()
+        .fold(0, |agg, entry| agg + entry.amount_stacked);
+
+    assert!(
+        participation <= liquid_ustx,
+        "CORRUPTION: More stacking participation than liquid STX"
+    );
+```
+
+The `RawRewardSetEntry` data are pulled directly from the
+`reward-cycle-pox-address-list` data map, and the `.amount_stacked` field is
+equal to the `total-ustx` field that was erroneously set in `stack-increase`.
+
+Considering these consequences with respect to the [blockchain catastrophic
+failure and recovery guidelines](https://github.com/stacksgov/sips/pull/10),
+this bug would be categorized as requiring an _immediate_ hard fork to rectify.
+THe network has not yet crashed, but it is in imminent danger of crashing and
+there is insufficient time to execute a fair, public SIP vote as has been customary for
+past hard forks. This SIP instead proposes that this hard fork activate at the
+start of the next reward cycle at the time of this writing.
+There is less than one reward cycle remaining to fix this
+problem, and yet a Stacker vote would require at least one complete reward cycle
+to carry out a STX coin vote (not accounting for any additional time required for
+sending out adequate public communications and tabulating the vote afterwards).
+
+# Specification
+
+Given the lack of time to conduct a fair, public vote to activate this SIP, the
+proposed fix in this SIP is as parsimonious and discreet as possible.  It will:
+
+* Disable `stack-increase` after the Bitcoin block activation height.  This
+  shall be achieved by altering the Clarity VM to declare that this function
+cannot be called, thereby preventing the buggy code body from being
+reached.  This shall not constitute a new version of the Clarity language;
+instead, the Stacks _chainstate specification_ shall be altered to require that
+this particular function in this particular contract after the Bitcoin block
+activation height is treated as absent from all Stacks forks.
+
+* Reset all `total-ustx` values in the `reward-cycle-pox-address-list` that
+  correspond to solo stackers to be equal to the amount of STX that is locked in
+their accounts.  This reset shall happen prior to any reward cycle processing in
+cycle #57 -- namely, prior to the call to
+`get_reward_threshold_and_participation()` and prior to calculating any STX
+auto-unlocks (which mutates state in `reward-cycle-pox-address-list`).
+
+  The routine for resetting `reward-cycle-pox-address-list` shall be invoked while
+  processing every Stacks block whose parent Stacks block was mined in a Bitcoin
+  block before the activation height.  Given that the set of solo stackers in
+  cycle #56 is small and fixed, the additional overhead of fixing all records in
+  `reward-cycle-pox-address-list` will be minimal.  Nevertheless, as a safety
+  precaution, each Stacks block whose parent was mined prior to the Bitcoin
+  block activation height must contain zero transactions and confirm zero
+  microblocks.
```

Comment:
Any transaction? No activity until the block activation?

Reply:
I read this as _no pox-2 transactions_, good question. Perhaps it can be made more explicit.

Reply:
+1 good to clarify -- I believe this is only relevant to the block in which the epoch activates.

Reply:
I read it as all blocks until activation. Perhaps to prevent potentially hitting the liquid Stacks assertion mentioned in the doc?

Reply:
The first Stacks block mined after the repairs to `reward-cycle-pox-address-list` are executed must contain zero transactions and confirm zero microblocks.  So, activity is allowed until the SIP activation, but the first Stacks block mined afterwards would be empty.